org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/concurrent/WriteRequest.java

```diff
@@ -42,18 +42,23 @@ public void run() {
 		try {
 			previous.join();
```

Comment:
I'd rename `previous` to `allPreviousRequestResults` or something to clarify that it is a future that will block until all previous requests have yielded a result (and hence all computation for these previous results is concluded).